### PR TITLE
Ensure project to work without mailer configuration

### PR DESCRIPTION
**Pull request by davydovanton**
I tried to solve https://github.com/hanami/hanami/issues/740, and I think my patch will work. But I don't understand how I can test this case. @jodosha (@hanami/core) could you help me with tests?

### CODE REVIEW

File: lib/hanami/components/components.rb [hunk 1]
```diff
@@ -177,9 +177,11 @@ module Components # rubocop:disable Metrics/ModuleLength
       end
 
       resolve do |configuration|
-        Hanami::Mailer.configuration = Hanami::Mailer::Configuration.new if Hanami.code_reloading?
-        Hanami::Mailer.configure(&configuration.mailer)
-        Hanami::Mailer.configuration
+        if configuration.mailer
```
Comment (AlfonsoUceda) [inline lib/hanami/components/components.rb]: @davydovanton I think should be like this line https://github.com/hanami/hanami/blob/9863c0949930f93155ed9362e2ea27bf732498c6/lib/hanami/components/components.rb#L105

**Comment by AlfonsoUceda** [comment]
@davydovanton thanks for this PR, to test this you should use as foundation this spec https://github.com/hanami/hanami/blob/master/spec/isolation/components/model_configuration/without_hanami_model_spec.rb and add to this folder https://github.com/hanami/hanami/tree/master/spec/isolation/components/mailer_configuration

can you try this? ;)

File: lib/hanami/components/components.rb (hunk 1 above)
Comment (mereghost) [inline lib/hanami/components/components.rb]: The trouble is that `mailer` is a hard dependency of `hanami`, unlike `model`.

If we ignore the configuration silently we may happen into issues complaining about it not working at all.

Either we make it a soft-dependency or put in a warn that strange stuff may happen.

Comment (davydovanton) [inline lib/hanami/components/components.rb]: I prefer to warn. @mereghost @AlfonsoUceda @jodosha WDYT?

**Comment by jodosha** [comment]
@davydovanton Please try with @AlfonsoUceda's suggestion. Create a new test file like this: https://github.com/hanami/hanami/blob/master/spec/isolation/components/mailer_configuration/with_hanami_mailer_spec.rb named `without_hanami_mailer_spec.rb`.

**Comment by davydovanton** [comment]
@AlfonsoUceda @jodosha thanks for help. I'll do it today 👍

File: spec/integration/cli/server_spec.rb [hunk 2]
```diff
@@ -364,4 +364,17 @@ def listing
       end
     end
   end
+
+  context "without email" do
+    it "returns page" do
+      with_project do
+        replace "config/environment.rb", "delivery :test", ""
```
Comment (jodosha) [inline spec/integration/cli/server_spec.rb]: @davydovanton I'm afraid this isn't enough. If you look at the original [bug report](https://github.com/hanami/hanami/issues/740#issue-211972961), there was more things removed. Specifically, all the `mailer` blocks were stripped from the original configuration. We should reproduce that scenario.

There are other examples here that show how to rewrite an entire file. That could be a solution.

File: spec/integration/cli/server_spec.rb [hunk 3]
```diff
@@ -364,4 +364,17 @@ def listing
       end
     end
   end
+
+  context "without email" do
```
Comment (jodosha) [inline spec/integration/cli/server_spec.rb]: s/without email/without malier/

File: spec/isolation/components/mailer_configuration/without_hanami_mailer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+RSpec.describe "Components: mailer.configuration", type: :cli do
+  context "without hanami-mailer" do
```
Comment (jodosha) [inline spec/isolation/components/mailer_configuration/without_hanami_mailer_spec.rb]: s/without hanami-mailer/without mailer configuration

File: spec/isolation/components/mailer_configuration/without_hanami_mailer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    it "resolves mailer configuration" do
+      with_project do
+        environment_file = Pathname.new(Dir.pwd).join("config", "environment")
+        replace "#{environment_file}.rb", "delivery :test", ""
```
Comment (jodosha) [inline spec/isolation/components/mailer_configuration/without_hanami_mailer_spec.rb]: Same of https://github.com/hanami/hanami/pull/742/commits/5523b86ca13a8f3145f3a41257b1b2c61f2b1f3b#r105839060

**Comment by davydovanton** [comment]
@jodosha thanks for help 💚

**Pull request merged**